src/component/registry/bpmn-elements-registry.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright 2020 Bonitasoft S.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ensureIsArray } from '../parser/json/converter/utils';
+import { BpmnMxGraph } from '../mxgraph/BpmnMxGraph';
+import { extractBpmnKindFromStyle } from '../mxgraph/style-helper';
+
+export function newBpmnElementsRegistry(graph: BpmnMxGraph): BpmnElementsRegistry {
+  return new BpmnElementsRegistry(new BpmnModelRegistry(graph), new HtmlElementRegistry(new BpmnQuerySelectors(graph.container?.id)));
+}
+
+/**
+ * @experimental subject to change, feedback welcome
+ */
+export class BpmnElementsRegistry {
+  constructor(private bpmnModelRegistry: BpmnModelRegistry, private htmlElementRegistry: HtmlElementRegistry) {}
+
+  // TODO doc, not found elements are not present in the return array
+  getElementsByIds(bpmnElementIds: string | string[]): BpmnElement[] {
+    // TODO move ensureIsArray to src/helpers/arrays.ts (not only for model) and add dedicated tests
+    return (ensureIsArray(bpmnElementIds) as string[])
+      .map(id => this.bpmnModelRegistry.getBpmnSemantic(id))
+      .filter(e => e)
+      .map(bpmnSemantic => ({ ...bpmnSemantic, htmlElement: this.htmlElementRegistry.getBpmnHtmlElement(bpmnSemantic.id) }));
+  }
+
+  // TODO we should also allow FlowKind
+  // we probably to introduce something like 'type BpmnKind = ShapeBpmnElementKind | FlowKind'
+  // getElementsByKinds(kinds: ShapeBpmnElementKind | ShapeBpmnElementKind[]): BpmnElement[] {
+  //   return [];
+  // }
+}
+
+interface BpmnSemantic {
+  id: string;
+  name: string;
+  /** `true` when relates to a BPMN Shape, `false` when relates to a BPMN Edge. */
+  isShape: boolean;
+  // TODO this would be more 'typed oriented' to use ShapeBpmnElementKind | FlowKind (as part of #929)
```

Comment:
```suggestion
  // TODO this would be more 'type oriented' to use ShapeBpmnElementKind | FlowKind (as part of #929)
  // we will probably introduce something like 'type BpmnKind = ShapeBpmnElementKind | FlowKind'
```